pkg/pillar/cmd/nodeagent/nodeagent.go

```diff
@@ -524,8 +531,14 @@ func handleLastRebootReason(ctx *nodeagentContext) {
 			if bootReason == types.BootReasonNone {
 				bootReason = types.BootReasonFirst
 			}
+		} else if previousSmartData.PowerCycleCount < smartData.PowerCycleCount {
+			reason = fmt.Sprintf("Reboot reason - power failure - at %s",
+				dateStr)
+			if bootReason == types.BootReasonNone {
+				bootReason = types.BootReasonPowerFail
+			}
 		} else {
-			reason = fmt.Sprintf("Unknown reboot reason - power failure or crash - at %s",
+			reason = fmt.Sprintf("Unknown reboot reason - crash - at %s",
```

Comment:
I think we should pick a more specific string based on whether the power cycle counter is -1 or not.
If it is -1 (no smartctl support) then keep current string. Otherwise the above crash string.
Also, I think there is a bootReason for the crash case.

Reply:
It makes sense to set it to BootReasonKernel in the crash case - could have been a hardware watchdog due to a hung kernel or a kernel panic.

Reply:
Changed